include/picongpu/particles/functor/User.hpp

```diff
@@ -49,7 +49,9 @@ namespace picongpu
                 template<typename DeferFunctor = Functor>
                 HINLINE User(
                     uint32_t currentStep,
-                    std::enable_if_t<std::is_constructible_v<DeferFunctor, uint32_t>>* = 0)
+                    std::enable_if_t<
+                        !std::is_default_constructible_v<
+                            DeferFunctor> && std::is_constructible_v<DeferFunctor, uint32_t>>* = 0)
```

Comment:
This line breaking looks very unintuitive since the AND is somewhere in the middle, but if the style check is fine with it, okay.